NitroxServer/Program.cs

```diff
@@ -58,5 +81,26 @@ static void configureCultureInfo()
             Thread.CurrentThread.CurrentCulture = cultureInfo;
             Thread.CurrentThread.CurrentUICulture = cultureInfo;
         }
+        
+        // See: https://docs.microsoft.com/en-us/windows/console/setconsolectrlhandler
+        static ConsoleEventDelegate consoleHandler;
+        private delegate bool ConsoleEventDelegate(int eventType);
+        [DllImport("kernel32.dll", SetLastError = true)]
+        private static extern bool SetConsoleCtrlHandler(ConsoleEventDelegate callback, bool add);
+        private static bool ConsoleEventCallback(int eventType)
+        {
+            if (eventType == 2)
+            {
+                Log.Info("Exiting ...");
+                Server.Instance.Stop();
+            }
+            return false;
+        }
+        private static void OnCtrlCPressed(object sender, ConsoleCancelEventArgs e)
+        {
+            Log.Info("Exiting ...");
+            Server.Instance.Stop();
```

Comment:
Can you add `Environment.Exit(0)` below this line? Otherwise it'll hang for me. I noticed it is because `Console.ReadLine()` is now waiting for text on a stream which is not receiving any text anymore because of Ctrl+C.

Reply:
I didnt notice that. will test it.

Reply:
You are right, this happens on windows. On linux it does not hang tho. Anyway I am adding an exit on it.